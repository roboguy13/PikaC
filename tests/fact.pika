%generate fact

Sll : layout[-x](List);
Sll Nil := emp;
Sll (Cons head tail) :=
  exists +nxt.
  x :-> head **
  (x+1) :-> nxt **
  Sll tail [nxt];


fact : Int -> Int;
fact n
  | n == 0 := 1;
fact n
  | not (n == 0) := n * (fact (n - 1));

%test "fact 5" Int: fact 5;

