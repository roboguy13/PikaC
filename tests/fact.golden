*** Running test fact 5
120

